doc/GenAIUsageGuidelines.md

```diff
@@ -0,0 +1,128 @@
+Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+version 2 with the GNU Classpath Exception [1] and GNU General Public
+License, version 2 with the OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] https://openjdk.org/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+-->
+
+# Eclipse OMR Generative Artificial Intelligence Usage Guidelines
+
+The Eclipse OMR project acknowledges that some contributions may be authored in
+whole or in part from generative artificial intelligence (GenAI) coding
+assistants.  Such contributions are permitted per the
+[Eclipse Foundation Generative AI Usage Guidelines](https://www.eclipse.org/projects/guidelines/genai/)
+which describes contributor and committer responsibilities, copyright and
+licensing considerations, and attribution requirements for GenAI assisted
+contributions.
+
+All contributors are encouraged to read and understand the advice and
+guidelines therein before submitting a contribution to Eclipse OMR even
+partially generated by an AI coding assistant.
+
+The Eclipse OMR project is subject to those policies and provides additional
+guidelines described below for contributors and committers governing
+GenAI-assisted contributions.
+
+## Attribution
+
+Contributions that include GenAI-assisted content must disclose that authorship
+in each file where GenAI content is present.  Contributors must include a
+statement like the following just below the copyright header at the beginning
+of the file that acknowledges their AI assistant (assuming such a statement
+does not already exist).
+
+```
+// Some sections generated by <YOUR AI ASSISTANT>
+```
+
+In addition, individual commits that include GenAI authored content must
+attribute that at the end of the commit message (before any human authorship
+signatures).
+
+```
+Co-authored by <YOUR AI ASSISTANT>
+```
+
+## Tagging
+
+Some contributors’ employers may require their AI-generated code and
+documentation contributions to be annotated with inline tags providing
+additional metadata about the generated content for legal and license
+attribution reasons. Eclipse OMR will accept contributions that include such
+metadata with some guidance below for contributors and committers.
+
+### Tag Usage
+
+Inline tagging of code or documentation generated by coding assistants is not
+mandatory.
+
+If tags are used, they must clearly demarcate the boundaries of GenAI-assisted
+content (e.g., via paired "begin" and "end" code comments). These boundaries
+may span consecutive lines, functions, paragraphs, or entire files. The term
+“assisted” is an important qualifier to describe the enclosed content because
+it may contain a blend of AI-generated content and human contributions. The
+“assisted” distinction is necessary because it may not be possible to clearly
```

Comment:
Nit:  the quote characters that surround "begin" and "end" are different from the quote character that surrounds “assisted”

Reply:
I agree that quoting should be consistent - my preference would be to use `"` consistently (unicode character 0x22, not the opening and closing quotes that appear around "assisted").

Reply:
Hm, thanks for spotting that.  I have no idea how that happened.  `"` was my intent, and I'll fix.